src/server/master_server/master_metadata_service_impl.cc

```diff
@@ -124,7 +126,34 @@ grpc::Status MasterMetadataServiceImpl::HandleFileCreation(
 grpc::Status MasterMetadataServiceImpl::HandleFileChunkRead(
     const protos::grpc::OpenFileRequest* request,
     protos::grpc::OpenFileReply* reply) {
-  return grpc::Status(grpc::StatusCode::UNIMPLEMENTED, "needs implementation");
+  // Step 1. Access the chunk_handle
+  const std::string& filename(request->filename());
+  const uint32_t chunk_index(request->chunk_index());
+
+  google::protobuf::util::StatusOr<std::string> chunk_handle_or(
+      metadata_manager()->GetChunkHandle(filename, chunk_index));
+
+  if (!chunk_handle_or.ok()) {
+    LOG(ERROR) << "Chunk handle not found: " 
```

Comment:
nit.  the error is not only chunk handle is not found; it can also be metadata manager failed to find the lock for it due to internal errors, so this error message isn't exactly correct

Reply:
I revised the error message